spec/rubocop/cop/lint/out_of_range_regexp_ref_spec.rb

```diff
@@ -8,39 +8,39 @@
   it 'registers an offense when references are used before any Regexp' do
     expect_offense(<<~RUBY)
       puts $3
-           ^^ Do not use out of range reference for the Regexp.
+           ^^ $3 is out of range (no Regexp capture groups detected).
     RUBY
   end
 
   it 'registers an offense when out of range references are used for named captures' do
     expect_offense(<<~RUBY)
       /(?<foo>FOO)(?<bar>BAR)/ =~ "FOOBAR"
       puts $3
-           ^^ Do not use out of range reference for the Regexp.
+           ^^ $3 is out of range (2 Regexp capture groups detected).
     RUBY
   end
 
   it 'registers an offense when out of range references are used for numbered captures' do
     expect_offense(<<~RUBY)
       /(foo)(bar)/ =~ "foobar"
       puts $3
-           ^^ Do not use out of range reference for the Regexp.
+           ^^ $3 is out of range (2 Regexp capture groups detected).
     RUBY
   end
 
   it 'registers an offense when out of range references are used for mix of numbered and named captures' do
     expect_offense(<<~RUBY)
       /(?<foo>FOO)(BAR)/ =~ "FOOBAR"
       puts $2
-           ^^ Do not use out of range reference for the Regexp.
+           ^^ $2 is out of range (1 Regexp capture groups detected).
```

Comment:
It'd be a bit better to say "1 group", but it's not a big deal.

Reply:
I agree but I'm not sure how to easily achieve this?

Reply:
Never mind I just implemented it directly, should be ok.